errors.go

```diff
@@ -0,0 +1,68 @@
+package netlink
+
+import (
+	"errors"
+	"fmt"
+	"os"
+)
+
+// Error messages which can be returned by Validate.
+var (
+	errMismatchedSequence = errors.New("mismatched sequence in netlink reply")
+	errMismatchedPID      = errors.New("mismatched PID in netlink reply")
+	errShortErrorMessage  = errors.New("not enough data for netlink error code")
+)
+
+// Errors which can be returned by a Socket that does not implement
+// all exposed methods of Conn.
+var (
+	errNotSupported = errors.New("operation not supported")
+)
+
+// notSupported provides a concise constructor for "not supported" errors.
+func notSupported(op string) *OpError {
+	return &OpError{
+		Op:  op,
+		Err: errNotSupported,
+	}
+}
+
+// IsNotExist determines if an error is produced as the result of querying some
+// file, object, resource, etc. which does not exist.  Users of this package
+// should always use netlink.IsNotExist, rather than os.IsNotExist, when
+// checking for specific netlink-related errors.
+//
+// Errors types created by this package, such as OpError, can be used with
+// IsNotExist, but this function also defers to the behavior of os.IsNotExist
+// for unrecognized error types.
+func IsNotExist(err error) bool {
+	switch err := err.(type) {
+	case *OpError:
+		// TODO(mdlayher): more error handling logic?
```

Comment:
> more error handling logic?

Only if it's netlink-specific, I imagine.

I will note that `os.IsNotExist` is good enough even if we end up using `*os.SyscallError` for the inner errors, since the `os` package deals with the unwrapping.

Reply:
TIL! To clarify though, this function is still required in order to unpack OpError.Err properly since it's a struct field right?  I assumed it made more sense to have our own version with these sorts of internal unpacking behaviors.

Reply:
Yes, that's right right. My comment about `os.IsNotExist` refers strictly to the inner error (the error in the `Err` field of `*OpError`).